recipes/inversify-cpp/all/test_package/conanfile.py

```diff
@@ -0,0 +1,20 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+    
+    def validate(self):
+        tools.check_min_cppstd(self, "17")
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```

Comment:
```suggestion
        if not tools.cross_building(self):
```

self.settings does not work when using build context.